frontend-react/src/components/Table/ColumnData.tsx

```diff
@@ -101,15 +102,29 @@ export const ColumnData = ({
             if (param) return action(rowData[param]);
             return action();
         };
+        const showActionButton = () => {
+            if (actionButtonHandler && actionButtonParam) {
+                return actionButtonHandler(rowData[actionButtonParam]);
+            } else {
+                return true;
+            }
+        };
+
         return tableData(
-            <Button
-                className="font-mono-2xs line-height-alt-4"
-                type="button"
-                unstyled
-                onClick={doAction}
-            >
-                {displayValue}
-            </Button>
+            <>
+                {showActionButton() ? (
```

Comment:
can `actionButtonHandler` or `actionButtonParam` ever actually be `undefined`? `actionButtonHandler` is just a simple `handleExpirationDate` function, and `actionButtonParam` is just a string that says `expires` to get the date. So we can just remove the `showActionButton()` and put `actionButtonHandler(rowData[actionButtonParam])` here?

or if it's dubious, just `actionButtonHandler?.(rowData[actionButtonParam])`

Reply:
actionButtonHandler and actionButtonParam are currently only being used in the DeliveriesTable. The other tables in the RS do not use these params and therefore can be undefined.

Reply:
but this exists within the `hasFeature("action")` block which should ensure that there is an action function. if that's not the case, then we should refactor what `hasFeature("action")` actually does.